python/llm/src/bigdl/llm/transformers/convert.py

```diff
@@ -57,22 +57,43 @@ def _replace_with_low_bit_linear(model, qtype, modules_to_not_convert=None,
             if not any(key in ".".join(current_key_name) for key in modules_to_not_convert):
                 with init_empty_weights():
-                    new_linear = LowBitLinear(
-                        module.in_features,
-                        module.out_features,
-                        qtype,
-                        module.bias is not None,
-                    )
+                    if qtype != "fp16":
+                        new_linear = LowBitLinear(
+                            module.in_features,
+                            module.out_features,
+                            qtype,
+                            module.bias is not None,
+                        )
 
-                    device_type = module.weight.data.device.type
-                    # Copy the weights
-                    paramsLowBit = FP4Params(data=module.weight.data,
-                                             requires_grad=False,
-                                             quantized=False,
-                                             _shape=None,
-                                             convert_shape_only=convert_shape_only,
-                                             qtype=qtype).to(device_type)
-                    new_linear._parameters['weight'] = paramsLowBit
+                        device_type = module.weight.data.device.type
+                        # Copy the weights
+                        paramsLowBit = FP4Params(data=module.weight.data,
+                                                 requires_grad=False,
+                                                 quantized=False,
+                                                 _shape=None,
+                                                 convert_shape_only=convert_shape_only,
+                                                 qtype=qtype).to(device_type)
+                        new_linear._parameters['weight'] = paramsLowBit
+                    else:
+                        # esimd fp16 path
+                        new_linear = FP16Linear(
+                            module.in_features,
+                            module.out_features,
+                            qtype,
+                            module.bias is not None,
+                        )
+                        device_type = module.weight.data.device.type
+                        #  only support two size now
+                        #  may generalize to other sizes
+                        if module.in_features in [4096, 11008]:
+                            new_linear.convert = True
+                            # convert here
+                            m, n = module.weight.data.shape
+                            trans_weight = module.weight.data.reshape(m//16, 16, n)
+                            trans_weight = trans_weight.transpose(1, 2).contiguous()
+                            new_linear._parameters['weight'] = nn.Parameter(trans_weight)
+                        else:
```

Comment:
if the size is not supported, there is no need to convert that layer

Reply:
for unsupported size, don't convert in else.

Reply:
Have revised this part, if the size is not supported, will keep original `Linear` unchanged.